functions/helloworld/hello-pub-sub/src/test/java/functions/ExampleSystemTest.java

```diff
@@ -0,0 +1,99 @@
+package functions;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.ProjectTopicName;
+import com.google.pubsub.v1.PubsubMessage;
+import io.github.resilience4j.core.IntervalFunction;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+import io.github.resilience4j.retry.RetryRegistry;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ExampleSystemTest {
+
+  // TODO<developer>: set these values (as environment variables, or hard-coded into your test)
```

Comment:
Let's not tell users to hard code configuration values into their code.

If you want to point them to something helpful, consider linking to SecretManager.

Reply:
Will drop the mention of hard-coding here. Based on our internal practices, environment variables should be an acceptable answer for this.